src-java/flowhs-topology/flowhs-storm-topology/src/test/java/org/openkilda/wfm/topology/flowhs/validation/FlowValidatorTest.java

```diff
@@ -231,46 +242,79 @@ public void checkForEncapsulationTypeRequirementCorrectTypeTest() throws Invalid
-                .build();
-    }
-
     @Test(expected = InvalidFlowException.class)
-    public void shouldFailIfMaxLatencyTier2HigherThanMaxLatency() throws InvalidFlowException {
+    public void failIfMaxLatencyTier2HigherThanMaxLatencyTest() throws InvalidFlowException {
         RequestedFlow flow = getTestRequestWithMaxLatencyAndMaxLatencyTier2((long) 1000, (long) 500);
         flowValidator.checkMaxLatency(flow);
     }
 
     @Test(expected = InvalidFlowException.class)
-    public void shouldFailIfMaxLatencyTier2butMaxLatencyIsNull() throws InvalidFlowException {
+    public void failIfMaxLatencyTier2butMaxLatencyIsNullTest() throws InvalidFlowException {
         RequestedFlow flow = getTestRequestWithMaxLatencyAndMaxLatencyTier2(null, (long) 500);
         flowValidator.checkMaxLatency(flow);
     }
 
     @Test
-    public void shouldNotFailIfMaxLatencyTier2andMaxLatencyAreNull() throws InvalidFlowException {
+    public void doesntFailIfMaxLatencyTier2andMaxLatencyAreNullTest() throws InvalidFlowException {
         RequestedFlow flow = getTestRequestWithMaxLatencyAndMaxLatencyTier2(null, null);
         flowValidator.checkMaxLatency(flow);
     }
 
     @Test
-    public void shouldNotFailIfMaxLatencyTier2andMaxLatencyAreEqual() throws InvalidFlowException {
+    public void doesntFailIfMaxLatencyTier2andMaxLatencyAreEqualTest() throws InvalidFlowException {
         RequestedFlow flow = getTestRequestWithMaxLatencyAndMaxLatencyTier2(500L, 500L);
         flowValidator.checkMaxLatency(flow);
     }
 
-    private static RequestedFlow buildFlow(int srcVlan, int dstVlan, Set<Integer> statVlans) {
+    @Test(expected = InvalidFlowException.class)
+    public void failOnAddingOneSwitchFlowToDiverseGroupWithExistingFlowTest() throws InvalidFlowException {
+        RequestedFlow oneSwitchFlow = buildOneSwitchFlow();
+        flowValidator.checkDiverseFlow(oneSwitchFlow);
+    }
+
+    @Test
+    public void doesntFailOnAddingFlowToDiverseGroupWithExistingOneSwitchFlowTest() throws InvalidFlowException {
+        RequestedFlow flow = RequestedFlow.builder()
+                .flowId(FLOW_1)
+                .srcSwitch(SWITCH_ID_1)
+                .srcPort(PORT_1)
+                .destSwitch(SWITCH_ID_2)
+                .destPort(PORT_1)
+                .detectConnectedDevices(new DetectConnectedDevices())
+                .diverseFlowId(DIVERSE_FLOW_ID)
+                .build();
+
+        Switch singleSwitch = Switch.builder().switchId(SWITCH_ID_1).build();
+
+        Flow oneSwitchFlow = Flow.builder()
+                .flowId("oneSwitchFlow")
+                .srcSwitch(singleSwitch)
+                .srcPort(PORT_1)
+                .destSwitch(singleSwitch)
+                .destPort(11)
+                .diverseGroupId(DIVERSE_FLOW_ID)
+                .build();
+
+        when(flowRepository.findById(DIVERSE_FLOW_ID)).thenReturn(Optional.of(oneSwitchFlow));
+        flowValidator.checkDiverseFlow(flow);
```

Comment:
It looks strange that there is no a single assertion. You can at least make sure that the flows exist. Alternatively, you can use `assertThatNoException()` idiom from assetj or you can add  `@Test(expected = None.class)` to make it explicit that no exception is expected.